File: awesome_places/lib/src/features/explore/data/models/place.dart
import 'dart:convert';
import 'dart:typed_data';

import 'address.dart';
import 'coordinate.dart';

class Place {
  String id;
  String name;
  String description;
  int placeType;
  Uint8List image;
  double rating;
  bool requiredFee;
  Coordinate? coordinate;
  Address? address;

  Place({
    required this.id,
    required this.name,
    required this.description,
    required this.placeType,
    required this.image,
    required this.rating,
    required this.requiredFee,
    required this.address,
    required this.coordinate,
  });

  factory Place.fromJson(Map<String, dynamic> json) {
    return Place(
        id: json['id'] as String,
        name: json['name'] as String,
        description: json['description'] as String,
        placeType: json['placeType'] as int,
        image: base64.decode(json['image']),
        rating: json['rating'] as double,
        requiredFee: json['requiredFee'] as bool,
        address: Address.fromJson(json['address']),
        coordinate: Coordinate.fromJson(json['coordinate']));
  }

  static List<Place>? fromListJson(List<dynamic> json) {
    List<Place> places = <Place>[];

    for (var element in json) {
      Place place = Place.fromJson(element);
      places.add(place);
    }
    return places;
  }
}

File: src/AwesomePlaces.Core/Entities/Place.cs
using AwesomePlaces.Core.Enums;
using AwesomePlaces.Core.Exceptions;

namespace AwesomePlaces.Core.Entities;

public class Place
{
	public Guid Id { get; protected set; } = Guid.NewGuid();
	private string Name { get; init; }
	private string Description { get; set; }
	private PlaceTypes PlaceType { get; init; }
	private byte[] Image { get; init; }
	private double Rating { get; init; }
	private bool RequiredFee { get; init; }
	public virtual Coordinate Coordinate { get; init; }
	public virtual Address Address { get; init; }

	private Place() { }

	public Place(Guid id, string name, string description, PlaceTypes placeType, Coordinate coordinate, Address address, byte[] image, double rating, bool requiredFee)
	{
		Id = id;
		Name = name;
		Description = description;
		PlaceType = placeType;
		Coordinate = coordinate;
		Address = address;
		Image = image;
		Rating = rating;
		RequiredFee = requiredFee;
	}

	public void ChangeDescription(string description)
	{
		if (string.IsNullOrWhiteSpace(description))
		{
			throw new InvalidPlaceDescriptionException(description);
		}

		this.Description = description;
	}
}
